WooCommerce/WooCommerceTests/ViewRelated/Products/Edit Product/ProductPriceSettingsViewModel+ProductVariationTests.swift

```diff
@@ -0,0 +1,82 @@
+
+        // Act
+        let sections = viewModel.sections
+
+        // Assert
+        let initialSections: [Section] = [
+            Section(title: Strings.priceSectionTitle, rows: [.price, .salePrice]),
+            Section(title: nil, rows: [.scheduleSale]),
+        ]
+        XCTAssertEqual(sections, initialSections)
+    }
+
+    func testTappingScheduleSaleToRowTogglesPickerRowInSalesSection() {
+        // Arrange
+        let saleStartDate: Date? = nil
+        let saleEndDate = Date()
+        let product = MockProductVariation().productVariation().copy(dateOnSaleStart: saleStartDate, dateOnSaleEnd: saleEndDate)
+        let viewModel = ProductPriceSettingsViewModel(product: product)
+        let initialSections: [Section] = [
+            Section(title: Strings.priceSectionTitle, rows: [.price, .salePrice]),
+            Section(title: nil, rows: [.scheduleSale, .scheduleSaleFrom, .scheduleSaleTo, .removeSaleTo]),
+        ]
+        XCTAssertEqual(viewModel.sections, initialSections)
+
+        // Act
+        viewModel.didTapScheduleSaleToRow()
+        let sectionsAfterTheFirstTap = viewModel.sections
+        viewModel.didTapScheduleSaleToRow()
+        let sectionsAfterTheSecondTap = viewModel.sections
+
+        // Assert
+        XCTAssertEqual(sectionsAfterTheFirstTap, [
+            Section(title: Strings.priceSectionTitle, rows: [.price, .salePrice]),
+            Section(title: nil, rows: [.scheduleSale, .scheduleSaleFrom, .scheduleSaleTo, .datePickerSaleTo, .removeSaleTo]),
+        ])
+        XCTAssertEqual(sectionsAfterTheSecondTap, initialSections)
+    }
+
+    func testRemovingSaleEndDateDeletesRemoveSaleToRow() {
+        // Arrange
+        let saleStartDate: Date? = nil
+        let saleEndDate = Date()
+        let product = MockProductVariation().productVariation().copy(dateOnSaleStart: saleStartDate, dateOnSaleEnd: saleEndDate)
+        let viewModel = ProductPriceSettingsViewModel(product: product)
+        let initialSections: [Section] = [
+            Section(title: Strings.priceSectionTitle, rows: [.price, .salePrice]),
+            Section(title: nil, rows: [.scheduleSale, .scheduleSaleFrom, .scheduleSaleTo, .removeSaleTo]),
+        ]
+        XCTAssertEqual(viewModel.sections, initialSections)
+
+        // Act
+        viewModel.handleSaleEndDateChange(nil)
+
+        // Assert
+        XCTAssertEqual(viewModel.sections, [
+            Section(title: Strings.priceSectionTitle, rows: [.price, .salePrice]),
+            Section(title: nil, rows: [.scheduleSale, .scheduleSaleFrom, .scheduleSaleTo]),
+        ])
+    }
+}
+
+private extension ProductPriceSettingsViewModel_ProductVariationTests {
+    enum Strings {
+        static let priceSectionTitle = NSLocalizedString("Price", comment: "Section header title for product price")
+        static let taxSectionTitle = NSLocalizedString("Tax Settings", comment: "Section header title for product tax settings")
+    }
```

Comment:
This is a bit of a problem, isn't it? 😁 What do you think about grouping all our localized strings into a single (`public`) file? I added that as a possible solution for https://github.com/woocommerce/woocommerce-ios/issues/2382. But maybe you have better ideas.

Also note that translations in unit tests are _captured_ to be translated in Glotpress. 😱 https://github.com/woocommerce/woocommerce-ios/issues/2229

Reply:
TIL'ed the translations in unit tests are also captured in the localization script! in this case, I moved all the localizable strings in `ProductPriceSettingsViewModel` to an enum extension that this test can access: https://github.com/woocommerce/woocommerce-ios/pull/2566/commits/9d121ed4533fb6ee7d23ea56f4a0a0b816f376ae

I also commented on https://github.com/woocommerce/woocommerce-ios/issues/2382 🙂